src/Microsoft.DotNet.ProjectModel/Resolution/ProjectDependencyProvider.cs

```diff
@@ -3,27 +3,48 @@
 
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
-using System.Runtime.Versioning;
 using Microsoft.Extensions.ProjectModel.Graph;
-using NuGet;
 using NuGet.Frameworks;
 
 namespace Microsoft.Extensions.ProjectModel.Resolution
 {
     public class ProjectDependencyProvider
     {
-        public ProjectDescription GetDescription(string name, string path, LockFileTargetLibrary targetLibrary)
+        public ProjectDescription GetDescription(string name,
+                                                 string path,
+                                                 LockFileTargetLibrary targetLibrary,
+                                                 Func<string, Project> projectCacheResolver)
```

Comment:
Should `projectCacheResolver` be something given to the `ProjectDependencyProvider` in the constructor? Is it likely to be different across invocations of `GetDescription`?